pkg/sync/expand.go

```diff
@@ -386,16 +386,19 @@ func (g *EntitlementGraph) mergeNodes(node1ID int, node2ID int) {
 	g.removeNode(node2ID)
 }
 
-func (g *EntitlementGraph) FixCycles() {
-	for {
+func (g *EntitlementGraph) FixCycles() error {
+	// If we can't fix the cycles in 10 tries, just give up
+	const maxTries = 10
+	for i := 0; i < maxTries; i++ {
 		cycles, hasCycles := g.GetCycles()
 		if !hasCycles {
-			return
+			return nil
 		}
 
 		// Merge all the nodes in a cycle.
 		for i := 1; i < len(cycles[0]); i++ {
 			g.mergeNodes(cycles[0][0], cycles[0][i])
 		}
 	}
+	return fmt.Errorf("could not fix cycles after %v tries", maxTries)
```

Comment:
feels like we might want slightly more information here on an error -- though i'm not quite sure what.

Reply:
We should have logged the cycles over in syncer.go before calling this function, so I also don't know what to put here.